File: 2021/day_1.rs
use std::fs;

fn load_data(filepath:&str) -> String {
  return fs::read_to_string(filepath).expect("Error!");
}

fn first_star() -> i32{
  let mut count:i32 = 0;
  let contents = load_data("input.txt");
  let vals = contents.split("\n").map(|a| {return a.parse::<i32>().unwrap();});
  let mut prev = 0;
  for num in vals {
    if num > prev {
      count = count + 1;
    }
    prev = num;
  };
  return count - 1;
}

fn second_star() -> i32{
  let data = load_data("input.txt");
  let contents = data.split("\n").map(|a| {return a.parse::<i32>().unwrap();}).collect::<Vec<_>>();
  let length = contents.len();
  let mut prev = 0;
  let mut count = 0;
  for i in 0..length-2 {
    let tot = contents[i] + contents[i+1] + contents[i+2];
    if tot > prev {
      count = count + 1;
    }
    prev = tot;
  }
  return count - 1;
}

fn main() {
  println!("First star: {}", first_star());
  println!("Second star: {}", second_star());
}

File: 2017/day_1.py
def load_input(source):
    with open(source) as f:
        data = [x.strip() for x in f.readlines()]
    if len(data) == 1:
        data = data[0]
    return data
    

def first_star(data):
    tot = 0
    for i in range(len(data) - 1):
        if data[i] == data[i+1]:
            tot += int(data[i])
    if data[-1] == data[0]:
        tot += int(data[0])
    return tot

def second_star(data):
    print(data)
    tot = 0
    step = len(data) // 2
    for i in range(len(data)):
        if data[i] == data[(i + step) % len(data)]:
            tot += int(data[i])
    return tot


def solution(source):
    data = load_input(source)
    print("Day 1")
    print("First star:", str(first_star(data)))
    print("Second star:", str(second_star(data)))
    print("-------------------------------------")


if __name__ == "__main__":
    solution("input.txt")
